app/Http/Controllers/LoanController.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Foundation\Auth\Access\AuthorizesRequests;
+use Illuminate\Foundation\Bus\DispatchesJobs;
+use Illuminate\Foundation\Validation\ValidatesRequests;
+use Illuminate\Routing\Controller as BaseController;
+use Illuminate\Http\Request;
+use App\Services\LoanService;
+
+class LoanController extends Controller
+{
+    protected $loanService;
+
+    public function __construct(LoanService $loanService)
+    {
+        $this->loanService = $loanService;
+    }
+
+    public function index()
+    {
+        return view('loan.store');
+    }
+
+    public function show(Request $request, int $id)
+    {
+        $data = $this->loanService->show($id);
+
+        return view('loan.view')->with('loan', $data);
+    }
+
+    public function store(Request $request)
+    {
+        $request->validate([
+            'loan_amount' => 'required|numeric|min:0',
+            'annual_interest_rate' => 'required|numeric|min:0',
+            'loan_term' => 'required|integer|min:1',
+            'extra_payment' => 'nullable|numeric|min:0'
+        ]);
+
+        $loanDetails = [
+            'loan_amount' => $request->input('loan_amount'),
+            'annual_interest_rate' => $request->input('annual_interest_rate'),
+            'loan_term' => $request->input('loan_term'),
+            'extra_payment' => $request->input('extra_payment'),
+        ];
```

Comment:
```suggestion
$loanDetails = $request->only([ 'loan_amount', 'annual_interest_rate', 'loan_term', 'extra_payment' )
```
Probably the same. I think this case is a better case.

Reply:
Yes, same but only seems to be better approach being the short form.